### PR TITLE
Implement continous integration practices + formatting and fixes across entire repo

### PR DESCRIPTION
**Pull request by kedhammar**

### CODE REVIEW

**Comment by kedhammar** [comment]
@remiolsen

File: anglerfish/__main__.py
Comment (kedhammar) [inline anglerfish/__main__.py:1]: This change made mypy stop complaining, but I'm a little bit scared of it 👀

File: anglerfish/demux/report.py [hunk 1]
```diff
@@ -85,47 +103,60 @@ def write_dataframe(self,outdir,samplesheet):
...
-                              "#reads", # We specify this for historical reasons
-                              "mean_read_len",
-                              "std_read_len",
-                              "i7_reversed",
-                              "i5_reversed",
-                              "ont_barcode"]
-
-
-
+    ont_barcode: str | None = None
```
Comment (kedhammar) [inline anglerfish/demux/report.py:153]: This might only be Python 3.10+ actually. Any chance of bumping the required version?

**Review by remiolsen — APPROVED** [review]
I most likely agree with all the checks made - ruff and the rules selected there especially. But that's a _me_  problem, not a problem with this PR 👍

**Pull request merged**